### PR TITLE
Add release channels add-accounts remove-accounts commands

### PR DESCRIPTION
**Pull request by sfc-gh-melnacouzi**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [ ] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Add commands to add and remove accounts from release channels:
`snow app release-channel add-accounts CHANNEL --target-accounts=org1.acc1,org2.acc2`
and
`snow app release-channel remove-accounts CHANNEL --target-accounts=org1.acc1,org2.acc2`

### CODE REVIEW

File: src/snowflake/cli/_plugins/nativeapp/entities/application_package.py [hunk 1]
```diff
@@ -861,6 +861,86 @@ def _bundle(self, action_ctx: ActionContext = None):
...
+                f"Release channel {release_channel} does not exist in application package {self.name}."
+            )
+
+        for account in target_accounts:
+            if not re.fullmatch(
+                f"{VALID_IDENTIFIER_REGEX}\\.{VALID_IDENTIFIER_REGEX}", account
+            ):
+                raise ClickException(
+                    f"Target account {account} is not in a valid format. Make sure you provide the target account in the format 'org.account'."
+                )
```
Comment (sfc-gh-gbloom) [inline src/snowflake/cli/_plugins/nativeapp/entities/application_package.py:895]: Let's extract this to a function?

**Review by sfc-gh-gbloom — COMMENTED** [review]
Did we finalize the command syntax? I would expect the channel to be specified as a parameter of the `release-channel` command, not the `add-accounts` sub-command:
```
snow app release-channel -c CHANNEL add-accounts --target-accounts=org1.acc1,org2.acc2
```

**Comment by sfc-gh-melnacouzi** [comment]
> Did we finalize the command syntax? I would expect the channel to be specified as a parameter of the `release-channel` command, not the `add-accounts` sub-command:
> 
> ```
> snow app release-channel -c CHANNEL add-accounts --target-accounts=org1.acc1,org2.acc2
> ```

Yes this was finalized in the design. 
Regarding your suggestion, I don't see any command currently using this pattern in SnowCLI.

the 2 contestants for this were:
`snow app release-channel add-accounts --channel=CHANNEL --target-accounts=org1.acc1,org2.acc2`
vs
`snow app release-channel add-accounts --target-accounts=org1.acc1,org2.acc2 CHANNEL`

I went with the second one to be consistent with release channels commands, and I also found similar pattern in `snow app version`. This was also decided based on feedback from the team on `snow app release-directive set DIRECTIVE` vs `snow app release-directive set --directive=directive`.

This is a bit trickier because of the composite verb "add-accounts".

File: src/snowflake/cli/_plugins/nativeapp/entities/application_package.py (hunk 1 above)
Comment (sfc-gh-melnacouzi) [reply src/snowflake/cli/_plugins/nativeapp/entities/application_package.py]: fixed

**Review by sfc-gh-gbloom — APPROVED** [review]

**Pull request merged**